plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/GitValueProviderUpdater.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.git.client;
+
+import org.eclipse.che.api.project.gwt.client.ProjectServiceClient;
+import org.eclipse.che.api.workspace.shared.dto.ProjectConfigDto;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.DtoUnmarshallerFactory;
+
+import javax.inject.Inject;
+
+import static org.eclipse.che.ide.api.notification.StatusNotification.Status.FAIL;
+
+/**
+ * Class for updating git attributes of current project
+ *
+ * @author Igor Vinokur
+ */
+public class GitValueProviderUpdater {
+    private final AppContext             appContext;
+    private final DtoUnmarshallerFactory dtoUnmarshallerFactory;
+    private final NotificationManager    notificationManager;
+    private final ProjectServiceClient   projectServiceClient;
+
+    @Inject
+    public GitValueProviderUpdater(AppContext appContext,
+                                   DtoUnmarshallerFactory dtoUnmarshallerFactory,
+                                   NotificationManager notificationManager,
+                                   ProjectServiceClient projectServiceClient) {
+        this.appContext = appContext;
+        this.dtoUnmarshallerFactory = dtoUnmarshallerFactory;
+        this.notificationManager = notificationManager;
+        this.projectServiceClient = projectServiceClient;
+    }
+
+    /**
+     * Update git attributes of current project in application context.
+     */
+    public void update(final ProjectConfigDto project) {
+        projectServiceClient.getProject(appContext.getWorkspace().getId(), project.getPath(), new AsyncRequestCallback<ProjectConfigDto>(
+                dtoUnmarshallerFactory.newUnmarshaller(ProjectConfigDto.class)) {
+            @Override
+            protected void onSuccess(ProjectConfigDto projectConfig) {
+                appContext.getCurrentProject().setRootProject(projectConfig);
```

Comment:
Use org.eclipse.che.ide.api.event.project.ProjectUpdatedEvent instead of setting root config into app context